File: BackendCitasMedicas/src/main/java/com/example/BackendCitasMedicas/Repository/PagoRepository.java
package com.example.BackendCitasMedicas.Repository;

import com.example.BackendCitasMedicas.Model.Pago;
import org.springframework.data.jpa.repository.JpaRepository;
import org.springframework.stereotype.Repository;

@Repository
public interface PagoRepository extends JpaRepository<Pago, Long> {
}

File: FrontendCitasMedicas/app/src/main/java/com/example/frontcitasmedicas/Repository/PagoRepository.kt
package com.example.frontcitasmedicas.Repository

import com.example.frontcitasmedicas.Model.Pago
import com.example.frontcitasmedicas.Service.RetrofitClient

class PagoRepository {

    private val api = RetrofitClient.pagoService

    suspend fun getPagos() = api.getAllPagos()

    suspend fun getPago(id: Long) = api.getPagoById(id)

    suspend fun addPago(pago: Pago) = api.addPago(pago)

    suspend fun updatePago(pago: Pago) = api.updatePago(pago)

    suspend fun deletePago(id: Long) = api.deletePago(id)
}
